src/gfx/reverse.cpp

```diff
@@ -241,24 +242,27 @@ void reverse() {
 
 	if (tilemap) {
 		if (attrmap) {
-			for (auto [id, attr] : zip(*tilemap, *attrmap)) {
-				bool bank = attr & 1 << 3;
-				if (id >= options.maxNbTiles[bank]) {
+			for (auto [tileID, attribute] : zip(*tilemap, *attrmap)) {
+				if (bool bank = attribute & 1 << 3; tileID >= options.maxNbTiles[bank]) {
```

Comment:
This doesn't take `options.baseTileIDs` into account.

Reply:
Yes, and that's the bug: you can't compare a tile ID to a number of tiles.

To make this clearer throughout the code, I'd suggest enforcing a naming convention “tile ID” for tilemap entries, as opposed to “tile offset” for indices into the tile data array.